pkg/syncer/resourcesync/controller.go

```diff
@@ -312,6 +314,21 @@ func (c *controller) process(ctx context.Context, key string) error {
 		c.notifySubscribers(ctx)
 	}
 
+	if err := c.PatchSyncTargetCondition(ctx, unauthorizedGVRs, syncTarget); err != nil {
+		errs = append(errs, err)
+	}
+
+	return errors.NewAggregate(errs)
+}
+
+type GVRAuthorizationChecker struct {
+	DownstreamKubeClient kubernetes.Interface
+	SyncTargetClient     workloadv1alpha1typed.SyncTargetInterface
```

Comment:
this hints at bad factoring if these must be public. They shouldn't.

Reply:
They were public in order to allow the e2e framework code to reuse them.
But you're right: that was ugly, and finally not as necessary as it initially seemed to be.

I completely removed `GVRAuthorizationChecker` as you suggested [here](https://github.com/kcp-dev/kcp/pull/2730#discussion_r1095689446)